src/olympia/addons/models.py

```diff
@@ -1323,6 +1324,28 @@ def promoted_group(self, *, group=None, application=None,
             not currently_approved or promoted.is_addon_currently_promoted)
         return promoted.group if is_promoted else NOT_PROMOTED
 
+    def promoted_applications(self):
+        """Returns a list of applications for which the addon is currently
+        promoted, or None if the addon is not currently promoted.
+        """
+        promoted_group = self.promoted_group()
+        if promoted_group != NOT_PROMOTED:
+            application = self.promotedaddon.application
+            if not application:
+                return [app.short for app in APP_USAGE]
+            return [application.short]
+        return None
+
+    @cached_property
+    def promoted(self):
+        promoted_group = self.promoted_group()
+        if promoted_group != NOT_PROMOTED:
+            return {
+                'category': promoted_group.api_name,
+                'applications': self.promoted_applications(),
```

Comment:
if `Addon.promoted_applications` is only used here maybe we should fold it all together in the same method so we don't have to get `self.promoted_group` twice.

`'applications': [self.promotedaddon.application] if self.promotedaddon.application else [app.short for app in APP_USAGE]` or similar?